client/tokenized-payment-request/frontend-utils.js

```diff
@@ -107,3 +126,112 @@ export const displayLoginConfirmationDialog = ( paymentRequestType ) => {
+ * and transforms it into the format expected by the Store API.
+ *
+ * @param {Object|null} cartDataCanMakePayment cartData from canMakePayment function.
+ * @param {Object|null} cartDataContent cartData from content component.
+ * @return {Object} Cart totals object.
+ */
+export const transformCartDataForStoreAPI = (
+	cartDataCanMakePayment,
+	cartDataContent
+) => {
+	let mappedCartData = {};
+
+	if ( cartDataCanMakePayment ) {
+		mappedCartData = {
+			...cartDataCanMakePayment,
+			items: cartDataCanMakePayment.cart.cartItems,
+			totals: cartDataCanMakePayment.cartTotals,
+			needs_shipping: cartDataCanMakePayment.cartNeedsShipping,
+			shipping_rates: cartDataCanMakePayment.cart.shippingRates,
+		};
+	}
+
+	if ( cartDataContent ) {
+		mappedCartData = {
+			items: cartDataContent.cartItems,
+			totals: constructCartDataContentTotals( cartDataContent ),
+			needs_shipping: cartDataContent.needsShipping,
+			shipping_rates: cartDataContent.shippingRates,
+			extensions: cartDataContent.extensions,
+		};
+	}
+
+	return mappedCartData;
+};
+
+function constructCartDataContentTotals( cartDataContent ) {
+	const totals = {
+		total_items: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_items',
+			'value'
+		)?.toString(),
+		total_items_tax: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_tax',
+			'value'
+		)?.toString(),
+		total_fees: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_fees',
+			'value'
+		)?.toString(),
+		total_fees_tax: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_fees',
+			'valueWithTax'
+		)?.toString(),
+		total_discount: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_discount',
+			'value'
+		)?.toString(),
+		total_discount_tax: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_discount',
+			'valueWithTax'
+		)?.toString(),
+		total_shipping: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_shipping',
+			'value'
+		)?.toString(),
+		total_shipping_tax: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_shipping',
+			'valueWithTax'
+		)?.toString(),
+		total_price: cartDataContent.cartTotal.value.toString(),
+		total_tax: getPropertyByKey(
+			cartDataContent.cartTotalItems,
+			'total_tax',
+			'value'
+		)?.toString(),
+		currency_code: cartDataContent.currency.code,
+		currency_symbol: cartDataContent.currency.symbol,
+		currency_minor_unit: cartDataContent.currency.minorUnit,
+		currency_decimal_separator: cartDataContent.currency.decimalSeparator,
+		currency_thousand_separator: cartDataContent.currency.thousandSeparator,
+		currency_prefix: cartDataContent.currency.prefix,
+		currency_suffix: cartDataContent.currency.suffix,
+	};
+
+	return totals;
+}
+
+function getPropertyByKey( obj, key, property ) {
+	const foundItem = obj.find( ( item ) => item.key === key );
+	return foundItem ? foundItem[ property ] : null;
+}
```

Comment:
```suggestion
/**
 * Searches object for matching key and returns corresponding property value from matched item.
 *
 * @param {Object} obj Object to search for key.
 * @param {string} key Key to match in object.
 * @param {string} property Property in object to return correct value.
 * @return {int|null} Value to return
 */
const getPropertyByKey = ( obj, key, property ) => {
	const foundItem = obj.find( ( item ) => item.key === key );
	return foundItem ? foundItem[ property ] : null;
};
```
Just rewriting this consistently with other functions in this file. Also, doc comment pls: I actually had to log `obj` to console before I could understand what this function was actually doing lol. 😅

Reply:
I meant to add doc comments to these and forgot. Thanks for the nudge! Your comments were good so I used them - [bcd0021](https://github.com/Automattic/woocommerce-payments/pull/9150/commits/bcd00217a15119ebb7d8718de1a03cc33430df0e)